config/types.go

```diff
@@ -512,6 +525,14 @@ func (g *GatewayRegistration) ToConsulType() *api.AgentService {
 	return result
 }
 
+func (g *GatewayRegistration) GetHealthCheckPort() int {
```

Comment:
This is a copy of the same func from the `AgentServiceConnectProxyConfig` type. We could make this a package-level func to avoid the duplication:

```go
func GetProxyHealthCheckPort(p int) int {
	if p != 0 {
		return p
	}
	return DefaultProxyHealthCheckPort
}
```

Not blocking. Just a suggestion.